site/content/contributing.md

```diff
@@ -1,3 +1,204 @@
+
+* **[CdkProjectHandler](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.Orchestration/CdkProjectHandler.cs)** - Contains all the logic to interact with CDK.
+
+* **[CdkAppSettingsSerializer](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.Orchestration/CdkAppSettingsSerializer.cs)** - It writes all the deployment settings into the `appsettings.json` file. This file is then deserialized into [IRecipeProps](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.Recipes.CDK.Common/RecipeProps.cs) and passed to the [CDK templates.](https://github.com/aws/aws-dotnet-deploy/tree/main/src/AWS.Deploy.Recipes/CdkTemplates)
+
+* **[Exceptions.cs](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.Orchestration/Exceptions.cs)** - Declares all exceptions that are thrown by the AWS.Deploy.Orchestration component
+
+* **[AWS.Deploy.Orchestration.UnitTests](https://github.com/aws/aws-dotnet-deploy/tree/main/test/AWS.Deploy.Orchestration.UnitTests)** - Contains unit tests for this component
+
+
+### [AWS.Deploy.Constants](https://github.com/aws/aws-dotnet-deploy/tree/main/src/AWS.Deploy.Constants)
+
+This is a shared project and contains various constants used throughout the codebase
+
+
+### [AWS.Deploy.CLI.IntegrationTests](https://github.com/aws/aws-dotnet-deploy/tree/main/test/AWS.Deploy.CLI.IntegrationTests)
+
+This contains the integration test suite
+
+
+### Server Mode
+
+To support IDEs using the AWS .NET Deploy tool the CLI can be launched in server mode. This will run the CLI as a server that exposes an API IDEs will be able to interact with to perform deployment activities. When the server mode is started the deploy tool will act as an ASP.NET Core application. Server mode will keep all of the deployment logic and rules inside the deploy tool CLI. IDEs are treated as frontends rendering the information that comes from the deploy tool and passing any of the information the user has specified into the deploy tool to validate and persist.
+
+**Debugging the .NET Deployment Tool via Server Mode**
+
+1. [Install](https://aws.amazon.com/visualstudio/) the AWS Toolkit for Visual Studio
+2. Open the following file in a text editor - `%localappdata%\AWSToolkit\PublishSettings.json`
+3. Under the `DeployServer` parent, add a new property `AlternateCliPath` that points to the `AWS.Deploy.CLI` executable. This instruct the Toolkit to launch the deploy tool from a custom location.
+
+**Example**
+```json
+{
+   "DeployServer":{
+      "AlternateCliPath":"C:\\code\\aws-dotnet-deploy\\src\\AWS.Deploy.CLI\\bin\\Release\\netcoreapp3.1\\AWS.Deploy.CLI.exe",
```

Comment:
nit: Point to the `net6.0` folder instead of `netcoreapp3.1`. .NET Core 3.1 goes out of support in a few months and we should have a documentation be as forward looking as possible.

Reply:
fixed